shared/request.ts

```diff
@@ -13,7 +15,19 @@ export async function request(
   path: `/${string}`,
   body?: Record<string, unknown>
 ) {
-  const response = await fetch(HOST + path, {
+  let host = HOST;
+
+  const existingCredentials = await Keychain.getGenericPassword();
+  if (existingCredentials) {
+    const settings = await getItem(
+      'settings',
+      {},
+      existingCredentials.password
+    );
+    host = settings.serviceURL ?? host;
+  }
+
+  const response = await fetch(host + path, {
```

Comment:
Can we cache this to memory so we don't continuously request and decrypt from the storage? Maybe only refresh when the app changes focus state?